Add message for GHC-64088 : Use of forall identifier (#447)
haskellfoundation/error-message-index: message-index/messages/GHC-64088/index.md

---
title: Use of "forall" as an identifier
summary: Use of "forall" as an identifier is discouraged
severity: warning
flag: -Wforall-identifier
introduced: 9.6.1
---

The string "forall" is not a keyword in the Haskell 2010 standard.

> It is therefore possible to declare a toplevel function in Haskell 2010 which uses "forall" as a name.

Reviewer: ```suggestion
It is therefore possible to declare a top-level function in Haskell 2010 which uses `forall` as a name.
```
The dash is for grammar, the backticks for style :-)

> As a result of the accepted [GHC proposal 281](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0281-visible-forall.rst) the string "forall" will become a reserved keyword in a future version of GHC, and for this reason GHC warns against the use of "forall" as an identifier.

Reviewer: ```suggestion
As a result of the accepted [GHC proposal 281](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0281-visible-forall.rst) the string `"forall"` will become a reserved keyword in a future version of GHC, so GHC warns against the use of `forall` as an identifier.
```

Minor style stuff - I'd merge without, but figured I'd present it here.
